distribution/sc-proxy-dex/src/wrapped_farm_token_merge.rs

```diff
@@ -0,0 +1,332 @@
+            .merge_locked_asset_tokens(OptionalArg::Some(BoxedBytes::from(ACCEPT_PAY_FUNC_NAME)))
+            .execute_on_dest_context_custom_range(|_, after| (after - 1, after))
+    }
+
+    fn merge_farm_tokens(
+        &self,
+        farm_contract: &Address,
+        tokens: &[WrappedFarmToken<Self::BigUint>],
+    ) -> GenericTokenAmountPair<Self::BigUint> {
+        if tokens.len() == 1 {
+            let token = tokens[0].clone();
+
+            return GenericTokenAmountPair {
+                token_id: token.attributes.farm_token_id,
+                token_nonce: token.attributes.farm_token_nonce,
+                amount: token.token_amount.amount,
+            };
+        }
+
+        for entry in tokens.iter() {
+            self.farm_contract_merge_proxy(farm_contract.clone())
+                .deposit_tokens(
+                    entry.attributes.farm_token_id.clone(),
+                    entry.attributes.farm_token_nonce,
+                    entry.token_amount.amount.clone(),
+                )
+                .execute_on_dest_context();
+        }
+
+        self.farm_contract_merge_proxy(farm_contract.clone())
+            .merge_farm_tokens(OptionalArg::Some(BoxedBytes::from(ACCEPT_PAY_FUNC_NAME)))
+            .execute_on_dest_context_custom_range(|_, after| (after - 1, after))
+    }
+
+    fn merge_farming_tokens(
+        &self,
+        tokens: &[WrappedFarmToken<Self::BigUint>],
+    ) -> SCResult<GenericTokenAmountPair<Self::BigUint>> {
+        if tokens.len() == 1 {
+            let first_token = tokens[0].clone();
+            let farming_amount = self.rule_of_three(
+                &first_token.token_amount.amount,
+                &first_token.attributes.farm_token_amount,
+                &first_token.attributes.farming_token_amount,
+            );
+
+            return Ok(GenericTokenAmountPair {
+                token_id: first_token.attributes.farming_token_id,
+                token_nonce: first_token.attributes.farming_token_nonce,
+                amount: farming_amount,
+            });
+        }
+
+        let farming_token_id = tokens[0].clone().attributes.farming_token_id;
+        let locked_asset_token_id = self.locked_asset_token_id().get();
+
+        if farming_token_id == locked_asset_token_id {
+            Ok(self.merge_locked_asset_tokens_from_wrapped_farm(tokens))
+        } else {
+            self.merge_wrapped_lp_tokens_from_farm(tokens)
+        }
+    }
+
+    fn merge_wrapped_lp_tokens_from_farm(
+        &self,
+        tokens: &[WrappedFarmToken<Self::BigUint>],
+    ) -> SCResult<GenericTokenAmountPair<Self::BigUint>> {
+        let mut wrapped_lp_tokens = Vec::new();
+
+        for token in tokens.iter() {
+            let wrapped_lp_token_amount = self.rule_of_three(
+                &token.token_amount.amount,
+                &token.attributes.farming_token_amount,
+                &token.attributes.farm_token_amount,
```

Comment:
Should the last two arguments be swapped?
```
self.rule_of_three(
                &token.token_amount.amount,
                &token.attributes.farm_token_amount,
                &token.attributes.farming_token_amount,
)
```

Reply:
Yes, fixed. Thank you for pointing out

Reply:
Resolved